app/components/dialog/index.tsx

```diff
@@ -126,11 +128,15 @@ class DialogComponent extends React.PureComponent<DialogContainerProps, {}> {
     }
   };
 
-  private handleSubmitNewCollection = async (params: PostCollectionParams) => {
+  private handleSubmitNewCollection = async (params: PostCollectionParams, targetPaperId?: number) => {
     const { dispatch } = this.props;
 
     this.validateCollection(params);
-    await dispatch(Actions.postNewCollection(params));
+    if (targetPaperId) {
```

Comment:
No need to use if statement because you make `postNewCollection` function get a paperId as an optional argument.